modin/experimental/core/storage_formats/pandas/native_query_compiler.py

```diff
@@ -0,0 +1,1283 @@
+    __format__ = _register_default_pandas(pandas.DataFrame.__format__)
+    __ge__ = _register_default_pandas(pandas.DataFrame.__ge__, squeeze_series=True)
+    __gt__ = _register_default_pandas(pandas.DataFrame.__gt__, squeeze_series=True)
+    __le__ = _register_default_pandas(pandas.DataFrame.__le__, squeeze_series=True)
+    __lt__ = _register_default_pandas(pandas.DataFrame.__lt__, squeeze_series=True)
+    __ne__ = _register_default_pandas(pandas.DataFrame.__ne__, squeeze_series=True)
+    __or__ = _register_default_pandas(pandas.DataFrame.__or__, squeeze_series=True)
+    __rand__ = _register_default_pandas(pandas.DataFrame.__rand__, squeeze_series=True)
+    __reduce__ = _register_default_pandas(
+        pandas.DataFrame.__reduce__, return_modin=False
+    )
+    __reduce_ex__ = _register_default_pandas(
+        pandas.DataFrame.__reduce_ex__, return_modin=False
+    )
+    __ror__ = _register_default_pandas(pandas.DataFrame.__ror__, squeeze_series=True)
+    __rxor__ = _register_default_pandas(pandas.DataFrame.__rxor__, squeeze_series=True)
+    __sizeof__ = _register_default_pandas(pandas.DataFrame.__sizeof__)
+    __xor__ = _register_default_pandas(pandas.DataFrame.__xor__, squeeze_series=True)
+    abs = _register_default_pandas(pandas.DataFrame.abs)
+    add = _register_default_pandas(_register_binary("add"))
+    all = _register_default_pandas(pandas.DataFrame.all)
+    any = _register_default_pandas(pandas.DataFrame.any)
+    apply = _register_default_pandas(pandas.DataFrame.apply)
+    apply_on_series = _register_default_pandas(pandas.Series.apply, is_series=True)
+    applymap = _register_default_pandas(pandas.DataFrame.applymap)
+    astype = _register_default_pandas(pandas.DataFrame.astype)
+    case_when = _register_default_pandas(pandas.Series.case_when)
+    cat_codes = _register_default_pandas(lambda ser: ser.cat.codes, is_series=True)
+    combine = _register_default_pandas(_combine, squeeze_series=True)
+    combine_first = _register_default_pandas(
+        lambda df, other: df.combine_first(other), squeeze_series=True
+    )
+    compare = _register_default_pandas(pandas.DataFrame.compare)
+    concat = _register_default_pandas(_concat)
+    conj = _register_default_pandas(
+        lambda df, *args, **kwargs: pandas.DataFrame(np.conj(df))
+    )
+    convert_dtypes = _register_default_pandas(pandas.DataFrame.convert_dtypes)
+    count = _register_default_pandas(pandas.DataFrame.count)
+    corr = _register_default_pandas(pandas.DataFrame.corr)
+    cov = _register_default_pandas(pandas.DataFrame.cov)
+    cummax = _register_default_pandas(pandas.DataFrame.cummax)
+    cummin = _register_default_pandas(pandas.DataFrame.cummin)
+    cumprod = _register_default_pandas(pandas.DataFrame.cumprod)
+    cumsum = _register_default_pandas(pandas.DataFrame.cumsum)
+    delitem = _register_default_pandas(_delitem)
+    df_update = _register_default_pandas(
+        pandas.DataFrame.update, in_place=True, df_copy=True
```

Comment:
Why do we need to operate on data copy here?

Reply:
It was initially added as in pandas query compiler df_update had [copy](https://github.com/arunjose696/modin/blob/arun-sqc/modin/core/storage_formats/pandas/query_compiler.py#L461)  but could figure out this was just due to some issue with arrow,

 Thus in nativeQueryCompiler copy wont be required in df_update so removing this argument(`df_copy`)  from the function definition.